src/parquet/column/scanner.h

```diff
@@ -108,21 +108,21 @@ class TypedScanner : public Scanner {
       levels_buffered_ = typed_reader_->ReadBatch(batch_size_, &def_levels_[0],
           &rep_levels_[0], values_, &values_buffered_);
 
-      // TODO: repetition levels
-
+      value_offset_ = 0;
       level_offset_ = 0;
       if (!levels_buffered_) {
         return false;
       }
     }
-    *def_level = def_levels_[level_offset_++];
-    *rep_level = 1;
+    *rep_level = descr()->is_repeated() ? rep_levels_[level_offset_] : 0;
+    *def_level = descr()->is_optional() ? def_levels_[level_offset_] : 1;
```

Comment:
Yea, you are right. I will fix this.
 I think this still passes because `def_level > max_definition_level()` and it becomes same as REQUIRED.